src/AppInstallerCommonCore/Manifest/ManifestValidation.cpp

```diff
@@ -234,4 +236,104 @@ namespace AppInstaller::Manifest
 
         return resultErrors;
     }
+
+    // Validate msix and msixbundle installer manifest
+    std::vector<ValidationError> ValidateMsixManifest(
+        const Msix::PackageVersion packageVersion,
+        const ManifestInstaller& installer,
+        Msix::MsixPackageManifestCache& msixManifestsCache,
+        bool treatErrorAsWarning)
+    {
+        std::vector<ValidationError> errors;
+        std::optional<Msix::OSVersion> installerMinOSVersion;
+
+        try
+        {
+            if (!installer.MinOSVersion.empty())
+            {
+                installerMinOSVersion = std::make_optional<Msix::OSVersion>(installer.MinOSVersion);
+            }
+        }
+        catch (const std::exception&)
+        {
+            errors.emplace_back(ManifestError::InvalidFieldValue, "MinimumOSVersion", installer.MinOSVersion);
+        }
+
+        std::vector<Msix::MsixPackageManifest> msixManifests;
+        try {
+            msixManifests = msixManifestsCache.GetAppPackageManifests(installer.Url);
+        }
+        catch (...)
+        {
+            errors.emplace_back(ManifestError::InstallerFailedToProcess, "InstallerUrl", installer.Url);
+        }
+
+        for (auto msixManifest : msixManifests)
+        {
+            // Validate package family name
+            auto msixManifestIdentity = msixManifest.GetIdentity();
+            auto msixPackageFamilyName = msixManifestIdentity.GetPackageFamilyName();
+            if (!installer.PackageFamilyName.empty() && !msixPackageFamilyName.empty())
+            {
+                if (installer.PackageFamilyName != msixPackageFamilyName)
+                {
+                    errors.emplace_back(ManifestError::InstallerMsixInconsistencies, "PackageFamilyName", msixPackageFamilyName);
+                }
+            }
+            // Yaml manifest missing package family name
+            else if (!msixPackageFamilyName.empty())

```

Comment:
I don't think this was full addressed.  Package family name is not an optional value in the manifest, so it shouldn't be optional from `msixManifestIdentity.GetPackageFamilyName()` and thus should never be empty.  If you want to ensure that, throw in `GetPackageFamilyName` and don't have code here that makes it seem like it could be.

Reply:
Pushed changes removing the condition checking for empty package family name from `msixManifestIdentity.GetPackageFamilyName()`